app/src/main/java/com/minimalisticapps/priceconverter/presentation/home/viewmodels/updateTextFieldModelWithFormatting.kt

```diff
@@ -0,0 +1,51 @@
+package com.minimalisticapps.priceconverter.presentation.home.viewmodels
+
+import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.text.input.TextFieldValue
+import com.minimalisticapps.priceconverter.common.utils.parseBigDecimalFromString
+import kotlin.math.max
+import kotlin.math.min
+
+fun updateTextFieldModelWithCommas(
```

Comment:
I have extracted the logic for formatting into a special function.

This logic was scattered in many  places now it is in this one method and can be tested!

This follows SRP, see: https://en.wikipedia.org/wiki/Single-responsibility_principle